### PR TITLE
chore: deprecate template create command in favor of template push

### LINKED ISSUE #9318: Fold template create into template push
Push is supposed to be a superset of create, so we're carrying significant code debt. The slight mismatches in their functionality are mostly mistakes, e.g.:
* create locally validates the template name whereas push does not
* create accepts TTLs whereas push does not

We should make sure to deprecate create for a period of time, perhaps with an annoying `time.Sleep`. We should also change `push` to create a new template by default.

Bonus points for reducing duplication across `template edit` too. See https://github.com/coder/coder/pull/9319 for an example bug.

### PR DESCRIPTION
**Pull request by f0ssel**
Closes https://github.com/coder/coder/issues/9318

What this changes:
- `coder template create` has been marked as deprecated and produces a warning pushing users to `push` and `edit` commands. 
- `--create` flag has been removed from `push` in favor of automatically creating if template does not exist. 
- `--private` flag has been added to `coder template edit` to meet parity with the previous `create` command flags. 
- `disable_everyone_group_access` bool has been added to the template edit API to support the new `--private` flag.
- Update docs to change mentions of `coder templates create` to `coder templates push`.

Examples:
```
➜  coder git:(f0ssel/templatepush-2) ✗ go run ./cmd/coder/main.go template create

DEPRECATION WARNING: This command will be removed in a future
release.
Use `coder templates push` command for creating and updating templates. 
Use `coder templates edit` command for editing template settings. 
```

```
➜  coder git:(f0ssel/templatepush-2) go run ./cmd/coder/main.go template -h   
coder v0.0.0-devel

USAGE:
  coder templates

  Manage templates

  Aliases: template

  Templates are written in standard Terraform and describe the infrastructure for workspaces
    - Make changes to your template, and plan the changes:
  
        $ coder templates plan my-template 
  
    - Create or push an update to the template. Your developers can update their
  workspaces:
  
        $ coder templates push my-template 

SUBCOMMANDS:
    archive     Archive unused or failed template versions from a given template(s)
    create      DEPRECATED: Create a template from the current directory or as specified by flag
    delete      Delete templates
    edit        Edit the metadata of a template by name.
    init        Get started with a templated template.
    list        List all the templates available for the organization
    pull        Download the active, latest, or specified version of a template to a path.
    push        Create or update a template from the current directory or as specified by flag
    versions    Manage different versions of the specified template

———
Run `coder --help` for a list of global options.
```

### CODE REVIEW

**Comment by ammario** [comment]
I'm good with this change as it gets us closer to the ideal state. We will have to mark the PR as a breaking change.

**Comment by f0ssel** [comment]
Great, I'll continue this one and mark it so. Thanks for the guidance on this one.

**Comment by f0ssel** [comment]
@ammario One more thing, since this will be a breaking change - should we just bite the bullet and remove the create command altogether? Or is that too drastic? I figure it may be better than breaking people twice, once now and once again when we remove it. (And lets us do the cleanup now instead of possibly forgetting to do it later).

**Comment by ammario** [comment]
That's a good point. We could go ahead and break "create" right now, but I don't think we should simply remove the command from the CLI as the error message would be scary. Instead, we can hide the subcommand and make it fail with something like "use `push` instead with the same arguments".

**Comment by ammario** [comment]
Hmm, I wonder if we can just make `create` an alias of `push`?

**Comment by f0ssel** [comment]
I went ahead and just made the command produce an error instructing users to `push` or `edit`, since an alias would just produce generic invalid flag errors without any other information.

**Comment by f0ssel** [comment]
@ammario @sreya I've made the changes we spoke about yesterday and this should be ready for review.

File: cli/templatecreate.go [hunk 1]
```diff
@@ -46,6 +42,15 @@ func (r *RootCmd) templateCreate() *clibase.Cmd {
 			r.InitClient(client),
 		),
 		Handler: func(inv *clibase.Invocation) error {
+			_, _ = fmt.Fprintln(inv.Stdout, "\n"+pretty.Sprint(cliui.DefaultStyles.Wrap,
```
Comment (sreya) [inline cli/templatecreate.go:45]: I'm not super familiar with the API of `clibase` but I think this would be better off implemented there so we can reuse it...maybe as a middleware?

I understand the intent of the `time.Sleep` but I think we should either make it something that users have to explicitly acknowledge via `stdin` (and thus are able to opt out of via `-y`) or we just accept that it's possible for the output to missed.

File: enterprise/cli/templatecreate_test.go [hunk 2]
```diff
@@ -1,143 +0,0 @@
-package cli_test
```
Comment (sreya) [inline enterprise/cli/templatecreate_test.go:1]: why were these deleted?

**Review by sreya — COMMENTED** [review]
It would be good if the command was marked as deprecrated in the `coder templates --help` output

File: enterprise/cli/templatecreate_test.go (hunk 2 above)
Comment (f0ssel) [reply enterprise/cli/templatecreate_test.go:1]: mistake from when I was removing the command

File: cli/templatecreate.go (hunk 1 above)
Comment (f0ssel) [reply cli/templatecreate.go:45]: Will look into the middleware idea

Yeah Ammar really liked the idea of the sleep, I'm fine removing it because I don't think it's worth forcing an input from the user.

**Review by sreya — APPROVED** [review]
LGTM, may be good to support deprecation in `clibase` but I don't have any suggestions for the best way to do it
